app/routes.js

```diff
@@ -30,13 +30,27 @@ const addContext = ({ context, req, csrfToken }) => ({
 
 export const routes = (authProvider, sessionManager) => {
   const router = express.Router();
+  const a = new Date();
+  const currentTimeInMillisecods = a.getTime();
 
   healthRoutes({ router, dependencies: getHealthCheckDependencies(config), logger });
 
   authenticationRoutes({
     router, authProvider, tokenType: 'id', logoutRedirectPath: config.logoutRedirectPath, logger,
   });
 
+  if (config.buyingcatalogueCookieConsentExpiration < currentTimeInMillisecods) {
+    router.use(async (req, res, next) => {
+      const cookieName = 'buyingcatalogue-cookie-consent';
+      const value = req.cookies[cookieName] ? JSON.parse(req.cookies[cookieName]) : undefined;
+      if (value && value.creationDate
+        && (value.creationDate < config.buyingcatalogueCookieConsentExpiration)) {
+        cookiePolicyClear({ res, logger });
+      }
+      return next();
+    });
+  }
+
```

Comment:
You could move router.use into buying-catalogue-library just to keep all the logic together. The error message is done in a similar way if you want to see an example.